### PR TITLE
Add daily build number to NuGet packages and fix package versioning.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This is to address issue #758.

This change adds BuildNumber.props containing the daily build number
property that's consumed by the package build infrastructure.

Updated package versions for Immutable and Metadata based on the values
of the daily building packages.  The values have a build number one
greater than the last build.

Deleted all nuspec files for non-shipping packages.

### CODE REVIEW

**Comment by mmitche** [comment]
@dotnet-bot test this please (restarted Jenkins while running)

File: src/dir.props [hunk 1]
```diff
@@ -1,3 +1,4 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="BuildNumber.props" />
   <Import Project="..\dir.props" />
```
Comment (weshaggard) [inline src/dir.props]: Is there any reason this is only under src/dir.props? Similar question for the BuildNumber.props why not put it in the root of the repo?

**Comment by weshaggard** [comment]
The changes look good to me but I want @nguerrera have a look at the version for Immutable/MetadataReader.

**Comment by nguerrera** [comment]
@weshaggard @jhendrixMSFT Those look fine as last shipped + 1, but we may switch to 1.0.0 (stable) before next ship.  

I saw something go buy for TestHost with a -4 at the end of the version number. I think the build number needs to be zero padded because in SemVer v1, it's a lexical comparison and you could get in to situations where foo-20 < foo-4. /cc @davkean

**Comment by jhendrixMSFT** [comment]
@nguerrera Thanks for pointing this out.  I assume we're using a 16-bit value for the daily build number just like we do for file versions.  I can update the build tools to pad this appropriately (e.g. 00004, 00123 etc).

**Comment by ellismg** [comment]
@jhendrixMSFT The Author for the commit is not correct.  Can you please ensure that your git user.name and user.email settings are correct, then reauthor the commit with git commit --amend --reset-author and then push the newer history?

**Comment by weshaggard** [comment]
@nguerrera very good point I forgot about that. @jhendrixMSFT we should get that fixed and once we have a build with the new version published on myget we should delete the existing packages from the myget feed.

**Comment by jhendrixMSFT** [comment]
I've updated the build tools package reference to the latest version as it includes the versioning changes brought up by @nguerrera.  Are there any other concerns?

**Comment by nguerrera** [comment]
LGTM.

**Comment by weshaggard** [comment]
@jhendrixMSFT You will need to also re-enable the building of packages in src\dirs.proj as I disabled it as part of the issue I filed. As is I don't expect that the packages are being built.

**Pull request merged**